### ISSUE TITLE
disable_eager_execution resets random seeds set before

### ISSUE THREAD

**Issue by bersbersbers** [tensorflow/tensorflow#35739]
This is on TF2.1 from pip on Windows 10.

**Describe the current behavior**
```
import tensorflow.compat.v1 as tf1

tf1.random.set_random_seed(0)
tf1.disable_eager_execution()

print(tf1.keras.backend.get_session().run(tf1.random.uniform((), 0, 1)))
```

This prints a different number every time. I have a similar example with more TF2-relevant code, where the same thing happens:
```
import tensorflow as tf
import tensorflow.compat.v1 as tf1

tf.random.set_seed(0)
tf1.disable_eager_execution()

print(tf1.keras.backend.get_session().run(tf.random.uniform((), 0, 1)))
```

A workaround is to set the seed after `disable_eager_execution`.

**Comment by ravikyram** [comment]
@bersbersbers 
Can you please try with 
```
import tensorflow as tf 
tf.random.set_seed(0)
print(tf.random.uniform((), 0, 1))
```
I am not seeing any issue with the above code. Thanks!

**Comment by bersbersbers** [comment]
@ravikyram you are right, the issue does not appear without `tf1.disable_eager_execution()`. This is why the title of this issue is "*disable_eager_execution resets random seeds* ...".

**Comment by ravikyram** [comment]
@bersbersbers 
I have tried on colab with TF version 2.1 and i am getting the same number after disabling the eager execution.Please, find the gist [here](https://colab.sandbox.google.com/gist/ravikyram/1f9c382caa98ab177d990c925f77a42d/untitled559.ipynb). Thanks!

**Comment by bersbersbers** [comment]
@ravikyram you are right, the issue does not appear when you set the seed *after* `tf1.disable_eager_execution()`. This is why the title of this issue is "... resets random seeds *set before*".

Setting the seed later is a possible as K have noted in in the original post. But the other way round, I consider it a bug.

**Comment by ravikyram** [comment]
@bersbersbers 

I tried other way around also. Earlier i missed in attaching the file. Please, find the gist [here](https://colab.sandbox.google.com/gist/ravikyram/9a093ecaa2305acc5cb3a9a8b2de6e2d/untitled562.ipynb).Please, let me know if i am missing something.Thanks!

**Comment by bersbersbers** [comment]
@ravikyram could you please stop changing the repro code and simply run the code examples that I have posted? In the latest gist, you entered `tf.compat.v1.disable_eager_execution` instead of `tf.compat.v1.disable_eager_execution()` - you are not calling this function. Also, the final line in the gist, `print(tf.random.uniform((), 0, 1))`, is not from my example code, either: in fact, it will fail once you correctly call `disable_eager_execution()`. Please, very simply, copy and paste this code and run it without modification, it will reproduce the issue:
```
import tensorflow.compat.v1 as tf1
tf1.random.set_random_seed(0)
tf1.disable_eager_execution()
print(tf1.keras.backend.get_session().run(tf1.random.uniform((), 0, 1)))
```

![image](https://user-images.githubusercontent.com/12128514/72334707-10194880-36be-11ea-97dd-dc46a6676c7d.png)
(Funny coincidence with the three very similar random numbers, though.)

**Comment by ravikyram** [comment]
I have tried on colab with TF version 2.1 and was able to reproduce the issue.Please, find the gist [here](https://colab.sandbox.google.com/gist/ravikyram/5637084fda4a8c4c84ea700e5a396d34/untitled565.ipynb). Thanks!

**Comment by wangpengmit** [comment]
Hi @bersbersbers, I think this is the intended behavior. Disabling eager execution is a very big change that will destroy many previous settings, including the global seed. `disable_eager_execution` should be the first thing you call before doing any TF-related stuff. 

BTW please see https://www.tensorflow.org/guide/random_numbers for the recommended ways to generate random numbers in TF2 eager mode.

**Comment by ccrusius** [comment]
Closing (intended behavior).
